allennlp/models/encoder_decoders/simple_seq2seq.py

```diff
@@ -1,23 +1,22 @@
 from typing import Dict, List, Tuple
 
 import numpy
-from overrides import overrides
 import torch
 import torch.nn.functional as F
+from overrides import overrides
 from torch.nn.modules.linear import Linear
-from torch.nn.modules.rnn import LSTMCell
 
 from allennlp.common.checks import ConfigurationError
 from allennlp.common.util import START_SYMBOL, END_SYMBOL
 from allennlp.data.vocabulary import Vocabulary
-from allennlp.modules.attention import LegacyAttention
-from allennlp.modules import Attention, TextFieldEmbedder, Seq2SeqEncoder
-from allennlp.modules.similarity_functions import SimilarityFunction
 from allennlp.models.model import Model
+from allennlp.modules import TextFieldEmbedder, Seq2SeqEncoder
+from allennlp.modules.seq2seq_decoders.decoder_cell import DecoderCell
+from allennlp.modules.seq2seq_decoders.simple_seq_decoder import SimpleSeqDecoder
 from allennlp.modules.token_embedders import Embedding
 from allennlp.nn import util
 from allennlp.nn.beam_search import BeamSearch
-from allennlp.training.metrics import BLEU
+from allennlp.training.metrics import BLEU, Metric
 
 
 @Model.register("simple_seq2seq")
```

Comment:
Instead of modifying this file directly, I'd just add a new implementation, like `generic_seq2seq` (or a better name...).  That way you don't need to worry about backwards compatibility of this class.

Reply:
Changed